globset/src/lib.rs

```diff
@@ -494,7 +494,7 @@ impl<'a> Candidate<'a> {
     /// Create a new candidate for matching from the given path.
     pub fn new<P: AsRef<Path> + ?Sized>(path: &'a P) -> Candidate<'a> {
         let path = path.as_ref();
-        let basename = file_name(path).unwrap_or(OsStr::new(""));
+        let basename = file_name(path).unwrap_or_else(|| OsStr::new(""));
```

Comment:
I don't agree with this change. The status quo is simpler and shorter.

Reply:
Can I ask why you disagree with it? It's a micro-optimization, but lazily creating the OsStr seems better to me. Eg: https://godbolt.org/g/YyXgWQ

Reply:
I don't disagree with it per se. I just think the change should be justified. If you can produce a useful benchmark in the context of `globset` that shows a meaningful difference, then I'd be happy to reconsider it.

Reply:
RE your godbolt link: that sort of evidence is _basically_ meaningless to me because it's measuring the wrong thing. The codegen for globset's code in particular would be better, and a real benchmark is best.